plugins/adrv9002.c

```diff
@@ -965,7 +963,20 @@ static void adrv9002_check_channel_status(struct plugin_private *priv,
 	 * widget as the channel is the same for all widgets on this port...
 	 */
 	ret = iio_channel_attr_read_double(chan->w[0].chn, "rf_bandwidth", &dummy);
-	if (ret == -ENODEV) {
+
+	if(ret < 0 && ret != -ENODEV) {
+		printf("Warning: iio channel returned an error when reading it: %d. We assume the channel is enabled\n",
+		       ret);
+	}
+
+	return ret != -ENODEV;
```

Comment:
I would probably do `ret == 0`. I would only consider a channel enabled if we get success. For the rf_bandwidth attr to return something rather than 0 or ENODEV (don't think the driver is returning any other ret value) likely means libiio is already somehow screwed (or some issue with the libiio backend) and odds are that things are far from being reliable.

I know this is not like it was before but now that I think about it, I think it makes more sense and I guess now is the time to change it.